sdk/communication/azure-communication-jobrouter/azure/communication/jobrouter/_generated/models/_patch.py

```diff
@@ -0,0 +1,141 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+"""Customize generated code here.
+
+Follow our quickstart for examples: https://aka.ms/azsdk/python/dpcodegen/python/customize
+"""
+
+try:
+    from typing import TypedDict  # pylint: disable=no-name-in-module
+except ImportError:
+    from typing_extensions import TypedDict
+
+from typing import List, Optional, Dict, Union, MutableMapping
+
+from datetime import datetime, timezone
+from dateutil.parser import parse
+from azure.core.serialization import _datetime_as_isostr  # pylint:disable=protected-access
+
+from ._models import (
+    JobQueue as JobQueueGenerated,
+    RouterWorker as RouterWorkerGenerated,
+    RouterJob as RouterJobGenerated,
+    JSON,
+)
+
+
+# cSpell:ignore tzinfos
+def _convert_str_to_datetime(
+        datetime_as_str: str
+) -> datetime:
+    dt = parse(datetime_as_str, tzinfos = [timezone.utc])
+    return dt
+
+
+QueueAssignment = TypedDict("QueueAssignment", {}, total = False)
+
+DeclineJobOfferResult = TypedDict("DeclineJobOfferResult", {}, total = False)
+
+ReclassifyJobResult = TypedDict("ReclassifyJobResult", {}, total = False)
+
+CancelJobResult = TypedDict("CancelJobResult", {}, total = False)
+
+CompleteJobResult = TypedDict("CompleteJobResult", {}, total = False)
+
+CloseJobResult = TypedDict("CloseJobResult", {}, total = False)
+
+
+class RouterJob(RouterJobGenerated):
+    def __init__(
+            self,
+            *,
+            notes: Dict[Union[str, datetime], str] = None,
+            labels: Optional[Dict[str, Union[int, float, str, bool, None]]] = None,
+            tags: Optional[Dict[str, Union[int, float, str, bool, None]]] = None,
+            **kwargs):
+
+        if notes:
+            for k in [key for key in notes.keys()]:
+                v: str = notes[k]
+                if isinstance(k, str):
+                    datetime_as_dt: datetime = _convert_str_to_datetime(k)    # pylint:disable=protected-access
+                    notes.pop(k)
+                    datetime_as_str: str = _datetime_as_isostr(datetime_as_dt)    # pylint:disable=protected-access
+                    notes[datetime_as_str] = v
+                elif isinstance(k, datetime):
+                    datetime_as_str: str = _datetime_as_isostr(k)    # pylint:disable=protected-access
+                    notes.pop(k)
+                    notes[datetime_as_str] = v
+
+        if labels:
+            for k, v in labels.items():
+                if not isinstance(v, (int, float, str, bool, type(None))):
+                    raise ValueError("labels only accept 'int', 'float', 'str', 'bool' and 'NoneType' as values.")
```

Comment:
What error does the service raise if we send tags or labels that aren't in the approved types?

Reply:
it will respond back with a 400 (bad request)

Reply:
With what kind of error message?
We should avoid this kind of client-side validation unless absolutely necessary.
If the service error message is useful, and helps customers understand what went wrong - then we should just let that throw.
If the service error is very unhelpful (e.g. "Malformed request"), then we can investigate how best to handle that.

Reply:
Currently the service simply throws a 400 with a generic message of "Failed to bind <property attribute>.", with a error code of "InvalidModelBindingError". We are still figuring out how to best handle this with a more meaningful message. 

So for this case, the error message will be something like "Failed to bind 'labels'." Currently we do not provide any information regarding which specific key-value pair in the dictionary is causing the error.

Reply:
In that case we can do this validation post-error, something like this:
```python
try:
    self._client.upsert_something(...)
except HttpResponseError as error:
    if error.error.code == "InvalidModelBindingError":
        if labels:
             # Do labels validation
             raise ValueError(...) from error
    raise

```

Reply:
Hmm... This feels a bit weird. I have removed this from the client side. Will get a service side fix in for this. In that way, it the error message will be uniform across all languages.